system/ee/ExpressionEngine/Service/Template/Variables/LegacyParser.php

```diff
@@ -264,7 +264,7 @@ public function extractVariables($tagdata, $target = null)
             // date variables
             elseif (strpos($val, 'format') !== false && preg_match("/.+?\s+?format/", $val)) {
                 $var_single[$val] = $this->extractDateFormat($val);
-            } else {  // single variables
+            } elseif (!is_numeric($val)) {  // single variables
                 $var_single[$val] = $val;
```

Comment:
Isn't an `else` missing here?

Reply:
No - we want to skip parsing if the variable is not matching any of rules